### PR TITLE
Update CONTRIBUTING.md prerequisites.

### PR DESCRIPTION
**Pull request by bdarnell**

### CODE REVIEW

**Comment by tschottdorf** [comment]
LGTM

**Comment by tamird** [comment]
:+1:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -3,9 +3,12 @@
...
 
 ### Assumed
- * A working GCC (on mac os x something like `xcode-select --install` will get you started)
- * [Go environment](http://golang.org/doc/code.html)
- * Git and Mercurial (for retrieving dependencies)
+ * A working C++ compiler (on mac os x something like `xcode-select
+   --install` will get you started). The compiler must support C++11
+   (GCC 4.9+ and clang 3.6+ are known to work).
+ * [Go environment](http://golang.org/doc/code.html). Currently a
+   64-bit version of go is required.
```
Comment (bdarnell) [inline CONTRIBUTING.md]: Vivek ran into this recently. The problem is that c-rocksdb fails to compile in 32-bit mode, although the issues are all warnings. If I remove -Werror from cgo_flags.go, it builds and `make test` passes, but I didn't investigate whether the errors were telling us anything we should be concerned about. I'm assuming we don't care enough about 32-bit builds to investigate any further and we should just declare them unsupported.

Comment (petermattis) [inline CONTRIBUTING.md]: Perhaps we should check for some of these preconditions in the makefile. For example, it wouldn't be difficult to check the Go version and whether it is 64-bit using `go version`. Similarly, we could determine what CC and CXX compiler Go is using (`go env CC CXX`) and verify they are recent enough.

**Pull request merged**